src/main/java/org/jlab/clas/std/services/convertors/EtRingToHipoReader.java

```diff
@@ -0,0 +1,293 @@
+                        try {
+                            byte[] data = mev.getData();
+                            evioBuffer.put(data, 0, mev.getLength());
+                        } catch (UnsupportedOperationException e) {
+                            evioBuffer.put(buf);
+                        }
+
+                        evioBuffer.order(buf.order());
+
+                        // ------------- EVIO
+//                        System.out.println("------> parsing event # " + mev +
+//                            " width length = " + mev.getLength());
+//                        try {
+                        EvioCompactReader reader = new EvioCompactReader(buf);
+                        ByteBuffer a = reader.getEventBuffer(1);
+                        EvioDataEvent eventEv = new EvioDataEvent(a, dict);
+
+                        // ----------- HIPO
+                        DataEvent decodedEvent = decoder.getDataEvent(eventEv);
+
+/*
+                        // get run number from the EtEvent
+                        HipoDataBank tmpBank = decoder.createHeaderBank(
+                            decodedEvent, -1, 10, torusField, solenoidField);
+                        int tmpRn = tmpBank.getInt("run", 0);
+                        // check if run number is changed
+                        if (tmpRn != runNumber) {
+                            runNumber = tmpRn;
+                            // go to RCDB and get torus and solenoid scales
+                            JDBCProvider
+                                provider = RCDB.createProvider("mysql://rcdb@clasdb.jlab.org/rcdb");
+                            provider.connect();
+
+                            solenoidField =
+                                (float) provider.getCondition(runNumber, "solenoid_scale").toDouble();
+                            torusField =
+                                (float) provider.getCondition(runNumber, "torus_scale").toDouble();
+                            provider.close();
+                            System.out.println("INFO: Processing conditions change. ======> runNumber = "
+                                + runNumber
+                                + ", torus = " + torusField
+                                + ", solenoid = " + solenoidField);
+                        }
+*/
+
+                        DataBank header = decoder.createHeaderBank(
+                            decodedEvent, runNumber, 10, torusField, solenoidField);
+
+                        decodedEvent.appendBanks(header);
+
+                        HipoDataEvent hipo = (HipoDataEvent) decodedEvent;
+                        evQueue.add(hipo.getHipoEvent());
+
+                    } catch (UnsupportedOperationException e) {
+                        System.out.println(e.getMessage());
+                    }
+                }
+
+                // put events back into ET system
+                sys.putEvents(att, mevs);
```

Comment:
If an exception is thrown, the events will not be put back into the ET system. That's why I had it on a `finally` block. But I don't know it the events are supposed to always be put back.

Reply:
That's was a good catch. events should always be put back.